contracts/zones/PausableZone.sol

```diff
@@ -0,0 +1,274 @@
+     *                    orders.
+     */
+    function executeMatchOrders(
+        SeaportInterface seaport,
+        Order[] calldata orders,
+        Fulfillment[] calldata fulfillments
+    )
+        external
+        payable
+        override
+        isOperator
+        returns (Execution[] memory executions)
+    {
+        // Call matchOrders on Seaport and return the sequence of transfers
+        // performed as part of matching the given orders.
+        executions = seaport.matchOrders{ value: msg.value }(
+            orders,
+            fulfillments
+        );
+    }
+
+    /**
+     * @notice Execute an arbitrary number of matched advanced orders,
+     *         each with an arbitrary number of items for offer and
+     *         consideration along with a set of fulfillments allocating
+     *         offer components to consideration components.
+     *
+     * @param seaport           The Seaport address.
+     * @param orders            The orders to match.
+     * @param criteriaResolvers An array where each element contains a reference
+     *                          to a specific order as well as that order's
+     *                          offer or consideration, a token identifier, and
+     *                          a proof that the supplied token identifier is
+     *                          contained in the order's merkle root.
+     * @param fulfillments      An array of elements allocating offer components
+     *                          to consideration components.
+     *
+     * @return executions An array of elements indicating the sequence of
+     *                    transfers performed as part of matching the given
+     *                    orders.
+     */
+    function executeMatchAdvancedOrders(
+        SeaportInterface seaport,
+        AdvancedOrder[] calldata orders,
+        CriteriaResolver[] calldata criteriaResolvers,
+        Fulfillment[] calldata fulfillments
+    )
+        external
+        payable
+        override
+        isOperator
+        returns (Execution[] memory executions)
+    {
+        // Call matchAdvancedOrders on Seaport and return the sequence of
+        // transfers performed as part of matching the given orders.
+        executions = seaport.matchAdvancedOrders{ value: msg.value }(
+            orders,
+            criteriaResolvers,
+            fulfillments
+        );
+    }
+
+    /**
+     * @notice Pause this contract, safely stopping orders from using
+     *         the contract as a zone. Restricted orders with this address as a
+     *         zone will not be fulfillable unless the zone is redeployed to the
+     *         same address.
+     */
+    function pause() external override isController {
+        // Emit an event signifying that the zone is paused.
+        emit Paused();
+
+        // Destroy the zone, sending any ether to the transaction submitter.
+        selfdestruct(payable(tx.origin));
```

Comment:
This lint rule either needs to be modified or (probably better) the behavior should be modified (maybe to take an address as an argument, or to have a configured recovery address)

Reply:
replaced with a passed in address argument